pkg/services/ppmshipment/ppm_shipment_creator.go

```
@@ -0,0 +1,86 @@
+		mtoShipmentCreator: mtoShipmentCreator,
+		checks: []ppmShipmentValidator{
+			checkShipmentID(),
+			checkPPMShipmentID(),
+			checkRequiredFields(),
+		},
+	}
+}
+
+// CreatePPMShipmentCheck passes a validator key to CreatePPMShipment
+func (f *ppmShipmentCreator) CreatePPMShipmentCheck(appCtx appcontext.AppContext, ppmShipment *models.PPMShipment) (*models.PPMShipment, error) {
+	return f.createPPMShipment(appCtx, ppmShipment, f.checks...)
+}
+
+func (f *ppmShipmentCreator) createPPMShipment(appCtx appcontext.AppContext, ppmShipment *models.PPMShipment, checks ...ppmShipmentValidator) (*models.PPMShipment, error) {
+	// Start a transaction that will create a Shipment, then create a PPM
+	transactionError := appCtx.NewTransaction(func(txnAppCtx appcontext.AppContext) error {
+		var err error
+		if ppmShipment.Shipment.ShipmentType == "" {
+			ppmShipment.Shipment.ShipmentType = models.MTOShipmentTypePPM
+		} else if ppmShipment.Shipment.ShipmentType != models.MTOShipmentTypePPM {
+			return apperror.NewInvalidInputError(uuid.Nil, nil, nil, "MTO shipment type must be PPM shipment")
+		}
+
+		if ppmShipment.Shipment.Status == "" {
+			ppmShipment.Shipment.Status = models.MTOShipmentStatusDraft
+		} else if ppmShipment.Shipment.Status != models.MTOShipmentStatusDraft {
+			return apperror.NewInvalidInputError(uuid.Nil, nil, nil, "Must have a DRAFT status")
+		}
+
+		if ppmShipment.Status == "" {
+			ppmShipment.Status = models.PPMShipmentStatusDraft
+		} else if ppmShipment.Status != models.PPMShipmentStatusDraft {
+			return apperror.NewInvalidInputError(uuid.Nil, nil, nil, "Must have a DRAFT status")
+		}
+
+		// NOTE: The ppm may require a service item for pricing.Passing an HHG service item may be sufficient for the pricer.
+		createShipment, err := f.mtoShipmentCreator.CreateMTOShipment(txnAppCtx, &ppmShipment.Shipment, nil)
+		// Check that mtoshipment is created. If not, bail out.
+		if err != nil {
+			return apperror.NewQueryError("MTOShipment", err, "")
```

I don't think we know that this was a QueryError, that to me would mean the database save failed.  I think there are a lot of different error types that can result from CreateMTOShipment so we should just return the err instead of wrapping it here.